### PR TITLE
Limit the Spark driver retry to 1 for debugging

### PR DESCRIPTION
**Pull request by wezhang**
For debugging senario, no need to retry the driver for failures.

Signed-off-by: Zhang Wei <zhwe@microsoft.com>

### CODE REVIEW

**Comment by vscjavaci** [comment]
Can one of the admins verify this patch?

**Comment by wezhang** [comment]
To fix #526

File: Utils/hdinsight-node-common/src/com/microsoft/azure/hdinsight/spark/common/SparkBatchRemoteDebugJob.java [hunk 1]
```diff
@@ -485,7 +485,9 @@ static SparkBatchRemoteDebugJob factory(
         driverOption = (carriedDriverOption.trim() + " " + driverDebugOption).trim();
         HashMap<String, Object> jobConfigWithDebug = new HashMap<>(submissionParameter.getJobConfig());
         SparkConfigures sparkConfigWithDebug = new SparkConfigures(sparkConf);
+
         sparkConfigWithDebug.put(sparkJobDriverJvmOptionConfKey, driverOption);
+        sparkConfigWithDebug.put("spark.yarn.maxAppAttempts", "1");
```
Comment (voidfunction) [inline Utils/hdinsight-node-common/src/com/microsoft/azure/hdinsight/spark/common/SparkBatchRemoteDebugJob.java]: I think we should have a way to make it visual for customers (on the debug panel?).  And let check the value is set or not first by customers's configuration file.

Comment (wezhang) [inline Utils/hdinsight-node-common/src/com/microsoft/azure/hdinsight/spark/common/SparkBatchRemoteDebugJob.java]: Here I clone a new instance for submission parameters, So, it won't effect the original customer's setting.

Comment (wezhang) [inline Utils/hdinsight-node-common/src/com/microsoft/azure/hdinsight/spark/common/SparkBatchRemoteDebugJob.java]: Fix it by throwing exception

**Comment by wezhang** [comment]
fix #526

**Pull request merged**